model/patch_lifecycle.go

```diff
@@ -524,18 +524,53 @@ func MakeCommitQueueDescription(patches []patch.ModulePatch, projectRef *Project
 	return "Commit Queue Merge: " + strings.Join(description, " || ")
 }
 
-func MakeMergePatchFromExisting(existingPatchID string) (*patch.Patch, error) {
-	existingPatch, err := patch.FindOne(patch.ById(patch.NewId(existingPatchID)))
+type EnqueuePatch struct {
+	PatchID string
+}
+
+func (e *EnqueuePatch) String() string {
+	return fmt.Sprintf("enqueue patch '%s'", e.PatchID)
+}
+
+func (e *EnqueuePatch) Send() error {
+	existingPatch, err := patch.FindOneId(e.PatchID)
 	if err != nil {
-		return nil, errors.Wrapf(err, "problem getting existing patch '%s'", existingPatchID)
+		return errors.Wrap(err, "can't get existing patch")
 	}
 	if existingPatch == nil {
-		return nil, errors.Errorf("patch '%s' doesn't exist", existingPatchID)
+		return errors.Errorf("no patch '%s' found", e.PatchID)
+	}
+
+	// only enqueue once
+	if len(existingPatch.MergePatch) != 0 {
+		return nil
+	}
+
+	mergePatch, err := MakeMergePatchFromExisting(existingPatch)
```

Comment:
Since this inserts a patch, can this be done after checking that there's a commit queue for the project?

Reply:
done